Some minor fixes of realtime logic

davidmz opened the pull request:
Previous version returned 'rooms' incorrectly in subscribe/unsubsribe acknowledgements

davidmz committed:
Call join/leave method in async way

Otherwise `buildGroupedListOfSubscriptions` builds rooms list before it is really changes

davidmz committed:
Use `noop` function from lodash

davidmz committed:
Export realtime event names and use them in pubsub-listener

davidmz committed:
Add test for subscribe/unsubscribe acknowledgements

davidmz committed:
Minor fix after rebase onto "stable"

indeyets commented on app/pubsub-listener.js:
```diff
@@ -76,13 +75,9 @@ export default class PubsubListener {
 
-      if (!isFunction(callback)) {
-        callback = noOp;
-      }
```
@davidmz Вот эта конструкция  тут была не случайна. Если клиент передаст параметром не колбэк, а что-то другое, то сервер упадёт. Нужно жёстко требовать чтобы это была функция

davidmz replied:
А разве в этом месте может быть что-то кроме функции или undefined? Ведь это не функция, которая на клиенте в браузере, а callback, который сам socket.io передаёт в обработчик. Тут (https://socket.io/docs/server-api/#socket-on-eventname-callback) вроде пишут, что это Function.

indeyets replied:
Увы, с точки зрения протокола это просто ещё один параметр. И клиент может передать лишний параметр со строкой, а колбэк не заказать. И будет проблема. Я тоже удивился

davidmz replied:
Клиент-то может… Давай я это потестирую, если действительно так, то, конечно, нужна будет проверка.

davidmz committed:
Design Session.sendAsync as a regular object method for uniformity

davidmz committed:
Smart handle socket events

This code is uniformly handles:
- callback parameter and it's absence
- debug logging on error
- Sentry exceptions capture
- acknowledgment messages

davidmz commented:
@indeyets да, ты был прав насчёт коллбэка, переписал там всё так, чтобы он всегда ловился правильно.
